### PR TITLE
fix: Subscribe to framework detection changes in wizard

### LINKED ISSUE #26388: CT framework detection can fail due to `open` mode not waiting for DataContext to initialize
### Current behavior

See `packages/server/lib/modes/index.ts` - this file kicks off an initialization of the data context for the current mode and executes a script based on the mode to launch the UI. The promise is passed into those scripts.
In `run.ts` the promise is awaited prior to reporting a "ready" state, but in `interactive.ts` it is not. This allows the UI to initialize prior to the data context being ready which can result in odd race conditions since not all GQL queries are reactive/subscriptions.

We need to evaluate whether an `await` is needed in `interactive.ts` to ensure this data is resolved prior to launching UI, or whether this is intentionally ignored for performance reasons

### Desired behavior

_No response_

### Test code to reproduce

1. Add an arbitrary delay to `packages/scaffold-config/src/detect.ts` in the `detectFramework` function of several seconds
2. Create a new CT project but do not initialize with a cypress config: `npm create vite@latest`. Install dependencies in project.
3. Open project directly into CT mode: `yarn dev --component --project <path to new project>`
4. Delay will prevent auto-detection of CT framework from resolving prior to UI being initialized

5. Add an `await` of the `_loading` promise in `interactive.ts`
6. Repeat test above, observe framework is now properly resolved

### Cypress Version

12.8.1

### Node version

16.17.1

### Operating System

any

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by mike-plummer**
- Closes #26388 

### Additional details
Launchpad does not wait for the data context to initialize in case there are significant slowdowns which would make the UI non-responsive on launch. This can cause the CT framework auto-detection to not populate if the detection is somehow delayed (complex project, slow machine, etc) - the detection occurs correctly but the GQL query in the Vue component has already resolved so the result is not reflected in the UI.

This PR adds a subscription so the Vue component can listen for updates to this data and reflect them in the UI.

### Steps to test
1. On `develop`, add a delay to wizard initialization by updating the `initialize` function with this line:
```
// WizardActions.ts
async initialize () {
  await new Promise((resolve) => setTimeout(resolve, 5000))
  ...
```
2. Create a new empty CT project: `npm create vite@latest`, use any supported framework & options
3. Install project dependencies and add Cypress: `cd <project dir>`, `npm i`, `npm i -D cypress`
3. Open project directly into CT mode: `cd <cypress repo> `, `yarn cypress:open --project <path to project> --component`
4. Observe that UI does not detect framework/bundler, stays on "Pick a framework"
5. Switch to this branch, repeat procedure
6. Observe framework & bundler are reflected in the UI after small delay

### How has the user experience changed?
**Before**

https://user-images.githubusercontent.com/11482842/230137927-c24db131-3603-4274-a3b7-d4fa7092d427.mov

**After**

https://user-images.githubusercontent.com/11482842/230137976-0895f4a2-2994-40f9-a8d1-2aea6081ca6d.mov

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by lmiller1990** [comment]
Mine shows up immediately regardless, even if I have the `setTimeout`. 🤔 Any ideas?


https://user-images.githubusercontent.com/19196536/230266100-421555fb-beb7-4ea4-8887-41b0f0dc8d16.mov

**Review by lmiller1990 — COMMENTED** [review]
Code looks good but I could not run the test plan - any ideas what I might be missing? Is there another step?

**Comment by mike-plummer** [comment]
> Code looks good but I could not run the test plan - any ideas what I might be missing? Is there another step?

@lmiller1990 Try `yarn dev` instead of `yarn cypress:open`? I was probably running against previously-built state when putting that test procedure together; IIRC `cypress:open` doesn't trigger any rebuild

**Review by jordanpowell88 — DISMISSED** [review]
The code looks good to me but the after video is less than ideal UX to me. Nothing indicates that we are doing work to search for your framework. Do we have a less disruptive way of alerting the user we are loading something after the initial paint?

**Comment by mike-plummer** [comment]
> The code looks good to me but the after video is less than ideal UX to me. Nothing indicates that we are doing work to search for your framework. Do we have a less disruptive way of alerting the user we are loading something after the initial paint?

@jordanpowell88 The five second delay is just to accentuate this scenario and make it more observable - in reality, this typically occurs within hundreds of milliseconds and occasionally happens to *just* miss the point where the gql query is resolved for the vue component. Unless the user is on a massively slow machine this shouldn't ever be observed as anything more than a "blip"

**Comment by jordanpowell88** [comment]
> > The code looks good to me but the after video is less than ideal UX to me. Nothing indicates that we are doing work to search for your framework. Do we have a less disruptive way of alerting the user we are loading something after the initial paint?
> 
> @jordanpowell88 The five second delay is just to accentuate this scenario and make it more observable - in reality, this typically occurs within hundreds of milliseconds and occasionally happens to _just_ miss the point where the gql query is resolved for the vue component. Unless the user is on a massively slow machine this shouldn't ever be observed as anything more than a "blip"

Ok that makes sense. I must have missed this context

**Comment by lmiller1990** [comment]
I tested locally, works great 💯

**Review by lmiller1990 — APPROVED** [review]

File: packages/server/lib/modes/interactive.ts [hunk 1]
```diff
@@ -158,6 +158,10 @@ export = {
   },
 
   async run (options: LaunchArgs, _loading: Promise<void>) {
+    // Note: We do not await the `_loading` promise here since initializing
+    // the data context can significantly delay initial render of the UI
+    // https://github.com/cypress-io/cypress/issues/26388#issuecomment-1492616609
```
Comment (emilyrohrbough) [inline packages/server/lib/modes/interactive.ts:162]: This is good context, but might be worth pointed to code vs a github comment in the event this is code ever changes so this stays "true" and/or relevant.

**Review by marktnoonan — APPROVED** [review]
Works well, and code looks good

**Pull request merged**